planner/core/rule_aggregation_push_down.go

```
@@ -299,91 +299,95 @@ func (a *aggregationPushDownSolver) pushAggCrossUnion(agg *LogicalAggregation, u
-			proj := a.convertAggToProj(newAgg)
+			proj, err := a.convertAggToProj(newAgg)
+			if err != nil {
+				return nil, err
+			}
 			proj.SetChildren(unionChild)
-			return proj
+			return proj, nil
 		}
 	}
 	newAgg.SetChildren(unionChild)
-	return newAgg
+	return newAgg, nil
 }
 
 func (a *aggregationPushDownSolver) optimize(p LogicalPlan) (LogicalPlan, error) {
 	if !p.context().GetSessionVars().AllowAggPushDown {
 		return p, nil
 	}
-	a.aggPushDown(p)
-	return p, nil
+	return a.aggPushDown(p)
 }
 
 // aggPushDown tries to push down aggregate functions to join paths.
-func (a *aggregationPushDownSolver) aggPushDown(p LogicalPlan) LogicalPlan {
+func (a *aggregationPushDownSolver) aggPushDown(p LogicalPlan) (_ LogicalPlan, err error) {
 	if agg, ok := p.(*LogicalAggregation); ok {
-		proj := a.tryToEliminateAggregation(agg)
-		if proj != nil {
-			p = proj
-		} else {
-			child := agg.children[0]
-			if join, ok1 := child.(*LogicalJoin); ok1 && a.checkValidJoin(join) {
-				if valid, leftAggFuncs, rightAggFuncs, leftGbyCols, rightGbyCols := a.splitAggFuncsAndGbyCols(agg, join); valid {
-					var lChild, rChild LogicalPlan
-					// If there exist count or sum functions in left join path, we can't push any
-					// aggregate function into right join path.
-					rightInvalid := a.checkAnyCountAndSum(leftAggFuncs)
-					leftInvalid := a.checkAnyCountAndSum(rightAggFuncs)
-					if rightInvalid {
-						rChild = join.children[1]
-					} else {
-						rChild = a.tryToPushDownAgg(rightAggFuncs, rightGbyCols, join, 1)
-					}
-					if leftInvalid {
-						lChild = join.children[0]
-					} else {
-						lChild = a.tryToPushDownAgg(leftAggFuncs, leftGbyCols, join, 0)
-					}
-					join.SetChildren(lChild, rChild)
-					join.SetSchema(expression.MergeSchema(lChild.Schema(), rChild.Schema()))
-					join.buildKeyInfo()
-					proj := a.tryToEliminateAggregation(agg)
-					if proj != nil {
-						p = proj
-					}
+		p, err = a.tryEliminateAgg(agg)
```

This error can be ignored?